src/cli/commands/init.ts

```diff
@@ -226,16 +230,93 @@ async function promptConfigSettings(disablePrompts: boolean, detectedConfig: Fra
       initial: detectedConfig.appDevserverUrl,
       format: trimValue,
     },
+    {
+      type: null,
+      name: "apiLanguage",
+    },
+    {
+      type: null,
+      name: "apiVersion",
+    },
   ]);
+
+  if (response.apiLocation) {
+    const apiLanguagePrompt = await prompts([
+      {
+        type: "select",
+        name: "apiLanguage",
+        message: "What's your API language? (optional)",
+        choices: [
+          { title: "Node.js", value: "node" },
+          { title: "Python", value: "python" },
+          { title: "Dotnet", value: "dotnet" },
+          { title: "Dotnet isolated", value: "dotnet-isolated" },
+        ],
+      },
+    ]);
+
+    response.apiLanguage = apiLanguagePrompt.apiLanguage;
+
+    const apiVersionPrompt = await prompts([
+      {
+        type: "select",
+        name: "apiVersion",
+        message: "What's your API version? (optional)",
+        choices: getChoicesForApiLanguage(response.apiLanguage),
+      },
+    ]);
+
+    response.apiVersion = apiVersionPrompt.apiVersion;
+  }
+
   return response;
 }
 
+function getChoicesForApiLanguage(apiLanguage: string) {
+  // Refer to this for functions and versions - https://learn.microsoft.com/en-us/azure/static-web-apps/configuration#selecting-the-api-language-runtime-version
+  let choices = [];
+  switch (apiLanguage) {
+    case "node":
+      choices = [
+        { title: "16", value: "16" },
+        { title: "14", value: "14" },
+        { title: "12", value: "12" },
+      ];
+      break;
+    case "python":
+      choices = [
+        { title: "3.8", value: "3.8" },
+        { title: "3.9", value: "3.9" },
+      ];
+      break;
+    case "dotnet":
+      choices = [
+        { title: "6.0", value: "6.0" },
+        { title: "3.1", value: "3.1" },
+      ];
+      break;
+    case "dotnet-isolated":
+      choices = [{ title: "6.0", value: "6.0" }];
+      break;
+    default:
+      choices = [
+        { title: "16", value: "16" },
+        { title: "14", value: "14" },
+        { title: "12", value: "12" },
+      ];
+      break;
```

Comment:
Note: find a better way for handling these languages and their versions. I am afraid this is gonna be challenging to maintain!

Reply:
Yes, currently this info is taken from SWA repo. Will figure out a better way to put this info in common place and use it in future iterations.